### PR TITLE
Unattended install: copy image from data directory to NFS shared directory

### PR DESCRIPTION
**Pull request by balamuruhans**
Tests using nfs setup (setup_local_nfs) expects image to be available in nfs shared path, which have to be taken care manually instead if image is available in data directory it can be copied and used.

### CODE REVIEW

File: virttest/tests/unattended_install.py [hunk 1]
```diff
@@ -1156,6 +1157,31 @@ def copy_images():
...
+                # check for image availability in images data directory
+                if(os.path.isfile(source) and not
+                   utils_misc.get_image_info(source)['lcounts'].lower() == "true"):
+                    logging.debug("Copying guest image from %s to %s", source,
+                                  dst)
+                    shutil.copy(source, dst)
+                else:
+                    raise exceptions.TestSetupFail("Guest image is unavailable"
+                                                   "/corrupted in %s and %s" %
+                                                   (source, dst))
```
Comment (nasastry) [inline virttest/tests/unattended_install.py]: According to `if` condition the message applies to `source dir` not the `destination dir`.

Comment (balamuruhans) [inline virttest/tests/unattended_install.py]: So logic here is, first it checks for image in NFS path, if not it checks in data directory, if image is not elsewhere then it reaches here. 
you can get it in `L1168`

Comment (nasastry) [inline virttest/tests/unattended_install.py]: Thanks for the clarification. @balamuruhans

**Comment by xutian** [comment]
Looks OK to me, ask @will-Do  to second review

**Comment by balamuruhans** [comment]
@nasastry @xutian Thank you for reviewing, @will-Do please help to review the PR

**Comment by balamuruhans** [comment]
**Tested:**
```
# avocado run --vt-type libvirt --vt-config /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/data/avocado-vt/backends/libvirt/cfg/migrate.cfg  --vt-only-filter  "Fedora.24.ppc64le virtio_scsi qcow2" --vt-no-filter "user network macvtap"
JOB ID     : e3ed4a5d6a97ad109614332fda40de3d9dfe8875
JOB LOG    : /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/results/job-2017-03-21T12.25-e3ed4a5/job.log
 (1/1) guest_import.qemu.qcow2.virtio_scsi.smp2.virtio_net.Fedora.24.ppc64le.powerkvm-qemu.unattended_install.import.import.default_install.aio_native: PASS (66.64 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
TESTS TIME : 66.64 s
JOB HTML   : /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/results/job-2017-03-21T12.25-e3ed4a5/html/results.html
```

**Code path:**
```
2017-03-21 12:25:23,506 unattended_insta L1173 DEBUG| Checking for image available in image data path - /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/data/avocado-vt/images/f24-ppc64le.qcow2
2017-03-21 12:25:23,506 process          L0386 INFO | Running 'qemu-img info /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/data/avocado-vt/images/f24-ppc64le.qcow2'
2017-03-21 12:25:23,519 process          L0476 DEBUG| [stdout] image: /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/data/avocado-vt/images/f24-ppc64le.qcow2
2017-03-21 12:25:23,519 process          L0476 DEBUG| [stdout] file format: qcow2
2017-03-21 12:25:23,520 process          L0476 DEBUG| [stdout] virtual size: 10G (10737418240 bytes)
2017-03-21 12:25:23,520 process          L0476 DEBUG| [stdout] disk size: 3.3G
2017-03-21 12:25:23,520 process          L0476 DEBUG| [stdout] cluster_size: 65536
2017-03-21 12:25:23,520 process          L0476 DEBUG| [stdout] Format specific information:
2017-03-21 12:25:23,520 process          L0496 INFO | Command 'qemu-img info /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/data/avocado-vt/images/f24-ppc64le.qcow2' finished with 0 after 0.0109951496124s
2017-03-21 12:25:23,520 process          L0476 DEBUG| [stdout]     compat: 1.1
2017-03-21 12:25:23,521 process          L0476 DEBUG| [stdout]     lazy refcounts: false
2017-03-21 12:25:23,521 process          L0476 DEBUG| [stdout]     refcount bits: 16
2017-03-21 12:25:23,521 process          L0476 DEBUG| [stdout]     corrupt: false
2017-03-21 12:25:23,521 unattended_insta L1178 DEBUG| Copying guest image from /var/lib/libvirt/images/workspace/runAvocadoFVTTest/avocado-fvt-wrapper/data/avocado-vt/images/f24-ppc64le.qcow2 to /home/bala/sharing/f24-ppc64le.qcow2
2017-03-21 12:25:46,035 unattended_insta L1015 INFO | Starting unattended install setup
```

File: virttest/tests/unattended_install.py [hunk 2]
```diff
@@ -1156,6 +1157,31 @@ def copy_images():
         process.system(dd_cmd)
     image_name = os.path.basename(dst)
     mount_point = params.get("dst_dir")
+
+    # libvirt import of guest from NFS shared path, then copy image
+    # from image_path to nfs mount dir
+    if(params.get("virt_test_type", "qemu") == "libvirt" and
+       params.get("setup_local_nfs", "no") == "yes"):
+        nfs_mount_path = params.get("nfs_mount_dir", None)
```
Comment (will-Do) [inline virttest/tests/unattended_install.py]: The default value 'None' is unnecessary

Comment (balamuruhans) [inline virttest/tests/unattended_install.py]: Okay, I will remove it.

**Comment by balamuruhans** [comment]
@will-Do done

**Comment by balamuruhans** [comment]
Thank you for review @sathnaga @xutian @will-Do @nasastry

**Pull request merged**